File: inheritance/java/SoccerMan.java
package schoo_Noro;

public class SoccerMan {

	//スタミナ
	private int stamina;
	//スピード
	private int pace;
	private int skill;

	public int get_skill(){
		return skill ;
	}

	SoccerMan(){
		this.stamina = 100;
		//スピード
		this.pace = 0;
	}

	public void set_skill(int new_skill){
		skill = new_skill;
	}

	public void get_status(){
		System.out.println("stamina:" + stamina);
		System.out.println("pace:" + pace);
		System.out.println("skill:" + skill);
	}
	public void run(){
		System.out.println(" -- run -- ");
		stamina--;
		pace++;
	}
	public void stop(){
		System.out.println(" -- stop -- ");
		pace--;
	}
	public void rest(){
		System.out.println(" -- rest -- ");
		stamina = 100;
	}

	public void acceleration(){
			System.out.println(" -- acceleration -- ");
			stamina--;
			stamina--;
			stamina--;
			pace = pace + 5;
		}

	public void sprint_speed(){
		System.out.println(" -- sprint_speed -- ");
		stamina--;
		stamina--;
		pace += skill;
	}
	public void practice(){
		System.out.println(" -- practice -- ");
		skill = skill + skill;
	}
}

File: inheritance/ruby/soccer_man.rb
class SoccerMan

  # スタミナ、スピード、スキル
  def initialize
	  @stamina = 100
		@pace = 0
		@skill = 0
  end

  def set_skill(new_skill)
    @skill = new_skill
  end

	def get_status
    puts("stamina:" + @stamina.to_s)
		puts("pace:" + @pace.to_s)
		puts("skill:" + @skill.to_s)
  end

	def run
    puts " -- run -- "
    @stamina -= 1
    @pace += 1
  end

	def stop
    puts " -- stop -- "
    @pace -= 1
  end

	def rest
    puts " -- rest -- "
    @stamina = 100
  end

	def acceleration
    puts " -- acceleration -- "
    @stamina -= 1
		@stamina -= 1
		@stamina -= 1
		@pace += 5
  end

  def sprint_speed
    puts " -- sprint_speed -- "
    @stamina -= 1
		@stamina -= 1
		@pace += @skill
	end

	def practice
    puts " -- practice -- "
    @skill += @skill
  end

end
